src/gui/optionsdlg.cpp

```diff
@@ -340,6 +341,7 @@ OptionsDialog::OptionsDialog(QWidget *parent)
     connect(m_ui->textWebUiUsername, &QLineEdit::textChanged, this, &ThisType::enableApplyButton);
     connect(m_ui->textWebUiPassword, &QLineEdit::textChanged, this, &ThisType::enableApplyButton);
     connect(m_ui->checkBypassLocalAuth, &QAbstractButton::toggled, this, &ThisType::enableApplyButton);
+    connect(m_ui->checkBypassAuthIPSubnet, &QAbstractButton::toggled, this, &ThisType::enableApplyButton);
```

Comment:
It might make sense to disable/enable the whitelist button when unchecking/checking the checkbox.

Also after editing the whitelist, the apply button should be enabled if the list has changed.

Reply:
I still haven't figured out how to do either of these things. Could you point me in the right direction?

Reply:
For the first one, use Qt signal/slots, connect qcheckbox toggled signal to qbutton setEnabled slot.
For the second, after closing edit dialog, if the list has changed then call enableApplyButton.

Reply:
Can you point to an existing example in the codebase that connects a qcheckbox and qbutton? I can't find how "Web User Interface", "Use HTTPS instead of HTTP", or any of the other checkboxes implement it.

Reply:
try
```c++
connect(m_ui->checkBypassAuthIPSubnet, &QCheckBox::toggled, m_ui->IPSubnetWhitelistButton, &QPushButton::setEnabled);
```
ps. I didn't test

Reply:
Thanks for the help. I've implemented both features for the whitelist. I've also updated the IP Ban List- in a separate commit- so it too implements the Apply button functionality.